src/test/java/org/elasticsearch/test/junit/rule/RepeatOnExceptionRule.java

```diff
@@ -55,9 +62,11 @@ public void evaluate() throws Throwable {
                     try {
                         base.evaluate();
                         return;
-                    } catch (BindTransportException t) {
-                        caughtThrowable = t;
-                        logger.info("Bind exception occurred, rerunning the test after [{}] failures", t, i+1);
+                    } catch (Throwable t) {
+                        if (t.getClass().equals(expectedException)) {
+                            caughtThrowable = t;
+                            logger.info("Bind exception occurred, rerunning the test after [{}] failures", t, i+1);
```

Comment:
is this still a bind exception all the time? If not we should update the error message